core/src/main/scala/caliban/parsing/adt/Selection.scala

```diff
@@ -3,8 +3,11 @@ package caliban.parsing.adt
 import caliban.InputValue
 import caliban.parsing.adt.Type.NamedType
 
+import scala.util.hashing.MurmurHash3
+
 sealed trait Selection extends Serializable {
-  @transient final override lazy val hashCode: Int = super.hashCode()
+  // TODO: Kept for binary compatibility
+  @transient override def hashCode: Int = super.hashCode()
```

Comment:
Another option is to make `Selection` extend `Product` in series/3.x and implement this here using `productHash` instead